src/Controller/Controller.php

```diff
@@ -572,6 +572,9 @@ public function render($view = null, $layout = null)
             return $this->response;
         }
 
+        // Re-fetch View class to pass view variables set in beforeRender callbacks
+        $this->View = $this->getView();
+
         if ($viewClass !== $this->viewClass) {
             $this->View = $this->getView($this->viewClass);
```

Comment:
Ah, I see. Well that can be very problematic. How about I simply set the new view variables just like the ```getView``` method does. Instead of calling the ```getView``` method. Thus keeping the configuration.